task-sdk/src/airflow/sdk/definitions/baseoperator.py

```diff
@@ -1555,6 +1555,46 @@ def render_template_fields(
             jinja_env = self.get_template_env()
         self._do_render_template_fields(self, self.template_fields, context, jinja_env, set())
 
+    def pre_execute(self, context: Any):
+        """Execute right before self.execute() is called."""
+        if self._pre_execute_hook is None:
+            return
+        from airflow.sdk.execution_time.callback_runner import create_executable_runner
+        from airflow.sdk.execution_time.context import context_get_outlet_events
+
+        create_executable_runner(
+            self._pre_execute_hook,
+            context_get_outlet_events(context),
+            logger=self.log,
+        ).run(context)
```

Comment:
Need to remove the `create_executable_runner` calls in task_runner so this is not wrapped twice. (Or alternatively we could just remove the call here.)

Reply:
Will remove

Reply:
Done